editor/components/block-list/style.scss

```diff
@@ -61,48 +57,77 @@
 		background-color: rgba( $white, 0.6 );
 	}
 
-	// simpler style for a block that has cursor focus (but hasn't been selected)
-	&.is-selected > .editor-block-mover:before,
-	&.is-hovered > .editor-block-mover:before,
-	&.is-selected > .editor-block-settings-menu:before,
-	&.is-hovered > .editor-block-settings-menu:before {
+
+	/**
+	 * Hovered Block style
+	 */
+
+	&.is-selected .editor-block-mover:before,
+	&.is-hovered .editor-block-mover:before,
+	&.is-selected .editor-block-settings-menu:before,
+	&.is-hovered .editor-block-settings-menu:before {
 		content: '';
 		position: absolute;
 		height: 36px;
 		top: 10px;
 	}
 
-	&.is-selected > .editor-block-mover:before,
-	&.is-hovered > .editor-block-mover:before {
+	&.is-selected .editor-block-mover:before,
+	&.is-hovered .editor-block-mover:before {
 		border-right: 1px solid $light-gray-500;
 		right: 6px;
 	}
 
-	&.is-reusable.is-selected > .editor-block-mover:before {
+	&.is-reusable.is-selected .editor-block-mover:before {
 		border-right: none;
 	}
 
-	&.is-selected > .editor-block-settings-menu:before,
-	&.is-hovered > .editor-block-settings-menu:before {
+	&.is-selected .editor-block-settings-menu:before,
+	&.is-hovered .editor-block-settings-menu:before {
 		border-left: 1px solid $light-gray-500;
 		left: 6px;
 	}
 
-	&.is-reusable.is-selected > .editor-block-settings-menu:before {
+	&.is-reusable.is-selected .editor-block-settings-menu:before {
 		border-left: none;
 	}
 
+
+	/**
+	 * Selected Block style
+	 */
+
+	.editor-block-list__block-edit {
```

Comment:
Do these styles need to be nested, or can they be defined as a separate top-level style block?

**Edit:** I guess it has some benefit to colocation of `&.is-selected` styles. Could also be separate, then targeting `.editor-block-list__block.is-selected & {`

Reply:
Not completely sure what you mean here, did you prefer the hover, and selected styles were moved out of `.editor-block-list__layout .editor-block-list__block`?

Reply:
`.editor-block-list__block-edit` - is unique, so it could be moved out of `.editor-block-list__layout` and everything should still work.

In case of:
```css
	// focused block-style
	&.is-selected .editor-block-list__block-edit:before {
		outline: 1px solid $light-gray-500;
	}
```
it would be more trickly, because of the dependency on the two parent classes.

Reply:
After Grzegorz' explanation (thank you), and further thinking about this, I understand your initial point here, but also reached your same notion that the colocation of the selected blocks styles is beneficial even though the code is slightly less optimal.

In that vein, I think the entire block-list could be cleaned up a bit, there's a lot bunched together in a less than ideal way. But it feels like that might be good to do separately, and get this PR in as is, especially given that this file is likely to be revisited soon with nested UI improvements and collapsing margins. 

Alternately we can move those rules out, as you suggest — it will just put them pretty far down the file as the `.editor-block-list__block` is pretty big as is.